src/lxcfs.c

```diff
@@ -895,28 +917,33 @@ static int lxcfs_flush(const char *path, struct fuse_file_info *fi)
 static int lxcfs_release(const char *path, struct fuse_file_info *fi)
 {
 	int ret;
+	enum lxcfs_virt_t type;
 
-	if (strncmp(path, "/cgroup", 7) == 0) {
+	type = file_info_type(fi);
+
+	if (LXCFS_TYPE_CGROUP(type)) {
 		up_users();
 		ret = do_cg_release(path, fi);
 		down_users();
 		return ret;
 	}
 
-	if (strncmp(path, "/proc", 5) == 0) {
+	if (LXCFS_TYPE_PROC(type)) {
 		up_users();
 		ret = do_proc_release(path, fi);
 		down_users();
 		return ret;
 	}
 
-	if (strncmp(path, "/sys", 4) == 0) {
+	if (LXCFS_TYPE_SYS(type)) {
 		up_users();
 		ret = do_sys_release(path, fi);
 		down_users();
 		return ret;
 	}
 
+	lxcfs_error("lxcfs_release() unknown file type: path=%s, type=%d",
```

Comment:
`lxcfs_error` already prints the function name from where it's called ;)

Reply:
Fixed